galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/NamespacesRoute.java

```diff
@@ -43,19 +44,20 @@ public NamespacesRoute(ResponseBuilder responseBuilder, IFramework framework ) {
 
     @Override
     public HttpServletResponse handleGetRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException, FrameworkException {
-        String namespaces = getNamespaces();
+        String namespaces = getNamespaces(req.getRequestURI());
 		return getResponseBuilder().buildResponse(response, "application/json", namespaces, HttpServletResponse.SC_OK); 
     }
 
-    private String getNamespaces() throws InternalServletException {
+    private String getNamespaces(String url) throws InternalServletException {
         logger.debug("Getting the list of namespaces");
-        JsonArray namespaceArray = new JsonArray();
+        List<Namespace> namespaceArray = new ArrayList<Namespace>();
         try {
             List<String> namespaces;
             namespaces = getFramework().getConfigurationPropertyService("framework").getCPSNamespaces();
             for (String name : namespaces) {
-                if ( !super.isHiddenNamespace(name) ) {
-                    namespaceArray.add(name);
+                Namespace namespace = new Namespace(name, url);
+                if ( !namespace.isHiddenNamespace() ) {
```

Comment:
Again, not sure if it makes sense to create a new Namespace object just to check if it's a hidden namespace - same applies throughout this PR.